python/ray/data/_internal/datasource/kafka_datasource.py

```diff
@@ -0,0 +1,568 @@
+"""Kafka datasource for bounded data reads.
+
+This module provides a Kafka datasource implementation for Ray Data that supports
+bounded reads with offset-based range queries.
+
+Message keys are decoded as UTF-8 strings (common case for routing keys).
+Message values are returned as raw bytes to support any serialization format
+(JSON, Avro, Protobuf, etc.). Users can decode values using map operations.
+
+Requires:
+    - kafka-python: https://kafka-python.readthedocs.io/
+"""
+
+import logging
+import time
+import uuid
+from typing import TYPE_CHECKING, Any, Dict, Iterable, List, Optional, Tuple, Union
+
+import pyarrow as pa
+
+if TYPE_CHECKING:
+    from kafka import KafkaConsumer, TopicPartition
+
+from ray.data._internal.output_buffer import BlockOutputBuffer, OutputBlockSizeOption
+from ray.data._internal.util import _check_import
+from ray.data.block import Block, BlockMetadata
+from ray.data.context import DataContext
+from ray.data.datasource import Datasource, ReadTask
+
+logger = logging.getLogger(__name__)
+
+
+def _build_consumer_config_for_discovery(
+    bootstrap_servers: List[str], authentication: Dict[str, Any]
+) -> Dict[str, Any]:
+    """Build minimal consumer config for partition discovery.
+
+    Args:
+        bootstrap_servers: List of Kafka broker addresses.
+        authentication: Authentication configuration dict.
+
+    Returns:
+        Consumer configuration dict for discovery.
+    """
+    config = {
+        "bootstrap_servers": bootstrap_servers,
+        "enable_auto_commit": False,
+        "auto_offset_reset": "latest",
```

Comment:
Can you explain what these parameters do and why they're set to these values? Wasn't obvious to me from reading